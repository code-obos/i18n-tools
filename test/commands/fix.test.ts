import { describe, it, expect, afterEach } from 'vitest';
import { vol } from 'memfs';
import { createLogger } from 'winston';
import Fix from '../../src/commands/fix';

describe('fix command', () => {
    afterEach(() => {
        vol.reset();
    });

    it('should create files missing for locales', () => {
        vol.fromNestedJSON(
            {
                'first_en.txt': 'first content',
                folder: {
                    'second_nb.txt': 'some content',
                },
            },
            '/app/messages',
        );

        Fix.run(
            createLogger(),
            {
                srcDir: '/app/messages',
            },
            false,
        );

        expect(vol.toJSON('/app/messages')).toMatchSnapshot();
    });

    it('should not exit if no errors are found', () => {
        vol.fromNestedJSON(
            {
                'first_en.txt': 'first content',
                'first_nb.txt': 'first content',
            },
            '/app/messages',
        );

        Fix.run(
            createLogger(),
            {
                srcDir: '/app/messages',
            },
            false,
        );

        expect(vol.toJSON('/app/messages')).toMatchSnapshot();
    });
});
