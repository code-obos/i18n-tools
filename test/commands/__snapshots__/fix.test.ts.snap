// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fix command > should create files missing for locales 1`] = `
{
  "/app/messages/first_en.txt": "first content",
  "/app/messages/first_nb.txt": "[nb] TODO",
  "/app/messages/folder/second_en.txt": "[en] TODO",
  "/app/messages/folder/second_nb.txt": "some content",
}
`;

exports[`fix command > should not exit if no errors are found 1`] = `
{
  "/app/messages/first_en.txt": "first content",
  "/app/messages/first_nb.txt": "first content",
}
`;
